src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java

```diff
@@ -1,31 +1,121 @@
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import java.util.Optional;
+import java.util.UUID;
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
 @ExtendWith(MockitoExtension.class)
 class PaymentServiceTest {
 
 
+    @Mock
+    private PaymentRepository paymentRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private ValidationService validationService;
+
+    @InjectMocks
+    private PaymentService paymentService;
+
+    @Captor
+    ArgumentCaptor<Payment> captor;
+
+    private static User user;
+    private static Payment payment;
+    private static Integer id;
+    private static Double amount;
+
     @BeforeEach
     void setUp() {
-
+        id = 1;
+        amount = 99.9;
+        user = new User(id, "Igor", Status.ACTIVE);
+        payment = new Payment(user.getId(), amount, "Payment from user " + user.getName());
     }
 
     @Test
     void createPayment() {
 
+        when(userRepository.findById(anyInt())).thenReturn(Optional.of(user));
+        when(paymentRepository.save(any(Payment.class))).thenReturn(payment);
+        Payment paymentResult = paymentService.createPayment(user.getId(), amount);
+
+        verify(validationService).validateUserId(user.getId());
+        verify(validationService).validateAmount(amount);
+        verify(validationService).validateUser(user);
+        verify(paymentRepository).save(captor.capture());
+
+        Payment captorResult = captor.getValue();
+
+        assertAll(
+                () -> assertEquals(payment, paymentResult),
+                () -> assertThat(paymentResult).isNotNull(),
+                () -> assertThat(captorResult.getMessage()).isEqualTo(paymentResult.getMessage()),
+                () -> assertThat(captorResult.getAmount()).isEqualTo(paymentResult.getAmount()),
+                () -> assertThat(captorResult.getUserId()).isEqualTo(paymentResult.getUserId())
+        );
     }
 
     @Test
     void editMessage() {
 
+        UUID uuid = payment.getPaymentId();
+        String message = "Mesaj";
+
+        when(paymentRepository.editMessage(uuid, message)).thenReturn(payment);
+        Payment editPayment = paymentService.editPaymentMessage(uuid, message);
+
+        verify(validationService).validatePaymentId(uuid);
+        verify(validationService).validateMessage(message);
+        verify(paymentRepository).editMessage(uuid, message);
+        assertEquals(payment, editPayment);
```

Comment:
verifyNoInteractions() for mocks that weren't used in this method also applicable to all other methods

Reply:
Done